pkg/daemon/ceph/osd/kms/vault.go

```diff
@@ -129,12 +129,25 @@ func configTLS(clusterdContext *clusterd.Context, namespace string, config map[s
 }
 
 func put(v secrets.Secrets, secretName, secretValue string, keyContext map[string]string) error {
+	// First we must see if the key entry already exists, if it does we do nothing
+	key, err := get(v, secretName, keyContext)
+	if err != nil {
+		// Because the error get wrapped in get(), we cannot compare 'err' against ErrSecretExists
+		if !strings.Contains(err.Error(), secrets.ErrInvalidSecretId.Error()) {
```

Comment:
Rook doesn't have control of this error message, right? What if it changes in the future? What if we stop wrapping the err in `get()`?

Reply:
Good point, let's remove the wrap.